### PR TITLE
Support engine parse for mysql

### PR DESCRIPTION
**Pull request by zihaoAK47**
Refer #25535.

Changes proposed in this pull request:
  - Add engine lexical rules
  - Modify engnameRef syntax rules
  - Add engine parsing and testing
  - Add create table option comment parse
  - Add comment parse test

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ✓] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ✓] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ✓] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ✓] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by strongduanmu** [comment]
Hi @zihaoAK47, thank you for your contribution recently. Can you send your social account to my email? We can talk about more advanced tasks.

**Comment by zihaoAK47** [comment]
> Hi @zihaoAK47, thank you for your contribution recently. Can you send your social account to my email? We can talk about more advanced tasks.

Error: Unable to find any artifacts for the associated workflow

Hi mentor, I executed the "./mvnw clean install - B - T1C - Dmaven. java doc. skip - Dmaven. jacoco. skip - e" test statement locally and all passed the test. However, I failed to check on Github.
Can I use WeChat for my social account? I have already sent it to your email and I am looking forward to participating in more advanced tasks. :laughing:

**Comment by zihaoAK47** [comment]
@strongduanmu Hi mentor please review thank

File: parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/Keyword.g4 [hunk 1]
```diff
@@ -108,14 +108,58 @@ AUTOCOMMIT
...
+    : N D B C L U S T E R
+    ;
+
+PERFORMANCE_SCHEMA
+    : P E R F O R M A N C E UL_ S C H E M A
+    ;
+
+TOKUDB
+    : T O K U D B
+    ;
```
Comment (strongduanmu) [inline parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/Keyword.g4:165]: Please add new line at the end of file.

File: parser/sql/dialect/mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/type/MySQLDDLStatementVisitor.java [hunk 2]
```diff
@@ -260,6 +262,13 @@ public ASTNode visitCreateTable(final CreateTableContext ctx) {
         if (null != ctx.createLikeClause()) {
             result.setLikeTable((SimpleTableSegment) visit(ctx.createLikeClause()));
         }
+        if (null != ctx.createTableOptions()) {
+            for (CreateTableOptionContext each : ctx.createTableOptions().createTableOption()) {
```
Comment (strongduanmu) [inline parser/sql/dialect/mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/type/MySQLDDLStatementVisitor.java:266]: Can you use visit call to reduce code level?

File: parser/sql/dialect/mysql/src/test/java/org/apache/shardingsphere/sql/parser/mysql/visitor/format/MySQLFormatVisitorIT.java [hunk 3]
```diff
@@ -76,23 +76,23 @@ public Stream<? extends Arguments> provideArguments(final ExtensionContext exten
                                     + "`runoob_test` NATIONAL CHAR(40),\n"
                                     + "`submission_date` DATE,\n"
                                     + "PRIMARY KEY (`runoob_id`)\n"
-                                    + ")ENGINE=InnoDB DEFAULT CHARSET=utf8;",
+                                    + ")ENGINE=INNODB DEFAULT CHARSET=utf8;",
```
Comment (strongduanmu) [inline parser/sql/dialect/mysql/src/test/java/org/apache/shardingsphere/sql/parser/mysql/visitor/format/MySQLFormatVisitorIT.java:79]: Why modify InnoDB to INNODB?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: parser/sql/dialect/mysql/src/test/java/org/apache/shardingsphere/sql/parser/mysql/visitor/format/MySQLFormatVisitorIT.java (hunk 3 above)
Comment (zihaoAK47) [reply parser/sql/dialect/mysql/src/test/java/org/apache/shardingsphere/sql/parser/mysql/visitor/format/MySQLFormatVisitorIT.java]: When modifying the vocabulary, this test did not pass, and now the previous modifications have been restored :blush:

**Comment by zihaoAK47** [comment]
@strongduanmu The modification has been completed please review

File: parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLCreateTableStatement.java [hunk 4]
```diff
@@ -46,4 +49,12 @@ public final class MySQLCreateTableStatement extends CreateTableStatement implem
     public Optional<SimpleTableSegment> getLikeTable() {
         return Optional.ofNullable(likeTable);
     }
+    
+    /**
+     * Get create table option segment.
```
Comment (strongduanmu) [inline parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLCreateTableStatement.java:54]: Please add new line after java doc.

File: test/it/parser/src/main/resources/case/ddl/create-table.xml [hunk 5]
```diff
@@ -51,8 +78,11 @@
         <column-definition type="varchar" start-index="39" stop-index="56">
             <column name="status" />
         </column-definition>
+        <create-table-option start-index="59" stop-index="95">
+            <engine name="InnoDB" start-index="66" stop-index="71" />
+        </create-table-option>
     </create-table>
-
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/case/ddl/create-table.xml:55]: Please keep same indent with previous line.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**